tests/test_shellcodes.py

```diff
@@ -0,0 +1,88 @@
+import random
+import re
+
+import archinfo
+import angr
+
+import pwnlib
+
+from angr.procedures.definitions import SimLibrary
+from angr.simos import SimLinux
+from archinfo import Endness
+from rex.exploit import Shellcodes
+
+
+arch_to_pwntools = {
+    'ARMEL': 'arm',
+    'MIPS32': 'mips',
+    'X86': 'i386',
+    'AMD64': 'amd64',
+}
+endness_to_pwntools = {
+    Endness.LE: 'little',
+    Endness.BE: 'big'
+}
+
+def test_dupsh(arch, fd_to_dup):
+    print(f"Testing shellcode to dup fd {fd_to_dup} for architecture {arch}!")
+    shellcode = Shellcodes['unix'][arch.name]['dupsh'](fd=(fd_to_dup,)).raw(arch=arch)
+    if arch.name == 'ARMEL':
+        # VEX sucks at decoding SVC instructions with operands that are non-zero, so we replace them
```

Comment:
I'd like to fix this on the vex level. can you comment on https://github.com/angr/angr/issues/2634?